apps/api/v2/src/modules/oauth/user/oauth-user.controller.ts

```diff
@@ -91,6 +92,31 @@ export class OAuthUserController {
     const user = await this.userRepository.update(userId, body);
     return { status: SUCCESS_STATUS, data: user };
   }
+
+  @Delete("/:userId")
+  @HttpCode(HttpStatus.OK)
+  @UseGuards(AccessTokenGuard)
+  async deleteUser(@Param("userId") userId: number): Promise<ApiResponse<Partial<User>>> {
+    this.logger.log(`Deleting user with ID: ${userId}`);
+
+    const existingUser = await this.userRepository.findById(userId);
```

Comment:
I think we should get the user through the oAuth client, make sure the user is associated to the client_id

Reply:
or just make sure this user has this oAuth client

Reply:
actually we should also check if the Access Token belongs to the  user you are trying to delete

Reply:
we should create a guard called, AccessTokenOwnershipGuard, this guard checks that the access token is tied to the userId

and we should update the AccessTokenGuard, it should check if the access_token belongs to the oauth client

Reply:
Done in https://github.com/calcom/cal.com/pull/12913